kernel/mempool.c

```diff
@@ -55,7 +55,8 @@ int k_mem_pool_alloc(struct k_mem_pool *p, struct k_mem_block *block,
 	__ASSERT(!(_is_in_isr() && timeout != K_NO_WAIT), "");
 
 	if (timeout > 0) {
-		end = z_tick_get() + _ms_to_ticks(timeout);
+		end = z_tick_get();
+		end += _ms_to_ticks(timeout);
```

Comment:
Wat?

Reply:
same thing :/

Reply:
Sorry, you're going to have to explain that to me.  What's undefined about that?  _ms_to_ticks() is a pure (in the functional sense) computation on the timeout argument.  z_tick_get() is a function call without side effects.  "end" is a local variable.  No change in computation order of this expression can cause a different result.

Reply:
Yes, I understood this particular case there is no side effect, though the evaluation of this is still not deterministic. I know that this is not a problem in our case but it is harmless. Writing a deviation for this seems to much for me